io-engine/src/bin/io-engine-client/v1/stats_cli.rs

```diff
@@ -235,6 +244,116 @@ async fn nexus(mut ctx: Context, matches: &ArgMatches) -> crate::Result<()> {
     Ok(())
 }
 
+async fn replica(mut ctx: Context, matches: &ArgMatches) -> crate::Result<()> {
+    ctx.v2("Requesting Replica metrics");
+    let replica_name = matches.get_one::<String>("name");
+    let response = ctx
+        .v1
+        .stats
+        .get_replica_io_stats(v1rpc::stats::ListStatsOption {
+            name: replica_name.cloned(),
+        })
+        .await
+        .context(GrpcStatus)?;
+    match ctx.output {
+        OutputFormat::Json => {
+            println!(
+                "{}",
+                serde_json::to_string_pretty(response.get_ref())
+                    .unwrap()
+                    .to_colored_json_auto()
+                    .unwrap()
+            );
+        }
+        OutputFormat::Default => {
+            let stats: &Vec<v1rpc::stats::ReplicaIoStats> =
+                &response.get_ref().stats;
+            if stats.is_empty() {
+                if let Some(name) = replica_name {
+                    ctx.v1(&format!(
+                        "No IoStats found for {}, Check if device exists",
+                        name
+                    ));
+                } else {
+                    ctx.v1("No Replica IoStats found");
+                }
+                return Ok(());
+            }
+
+            let table = stats
+                .iter()
+                .map(|p| {
+                    let io_stat = p.stats.as_ref().unwrap();
+                    let read_latency = ticks_to_time(
```

Comment:
I wonder if this can be captured in a closure, and use it below on the vec as such:
vec![ticks_to_time(io_state.write_latency_ticks), ticks_to_time(io_state.read_latency_ticks), ...

Reply:
Incorporated